paddle/fluid/memory/allocation/cpu_allocator.cc

```diff
@@ -23,6 +23,14 @@ namespace allocation {
 CPUAllocation::CPUAllocation(void *ptr, size_t size)
     : Allocation(ptr, size, platform::CPUPlace()) {}
 
+void CPUAllocation::share_data_with(void *ptr, size_t size) {
```

Comment:
Numpy Allocation should be an independent Allocation. Not a CPUAllocation

Reply:
Yes, maybe I will implement a independent one which is based on the CPUAllocation